state/aws/dynamodb/dynamodb.go

```diff
@@ -0,0 +1,168 @@
+package dynamodb
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbiface"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/dapr/components-contrib/state"
+)
+
+// StateStore is a DynamoDB state store
+type StateStore struct {
+	client dynamodbiface.DynamoDBAPI
+	table  string
+}
+
+type dynamoDBMetadata struct {
+	Region       string `json:"region"`
+	AccessKey    string `json:"accessKey"`
+	SecretKey    string `json:"secretKey"`
+	SessionToken string `json:"sessionToken"`
+	Table        string `json:"table"`
+}
+
+// NewDynamoDBStateStore returns a new dynamoDB state store
+func NewDynamoDBStateStore() *StateStore {
+	return &StateStore{}
+}
+
+// Init does metadata and connection parsing
+func (d *StateStore) Init(metadata state.Metadata) error {
+	meta, err := d.getDynamoDBMetadata(metadata)
+	if err != nil {
+		return err
+	}
+
+	client, err := d.getClient(meta)
+	if err != nil {
+		return err
+	}
+
+	d.client = client
+	d.table = meta.Table
+	return nil
+}
+
+// Get retrieves a dynamoDB item
+func (d *StateStore) Get(req *state.GetRequest) (*state.GetResponse, error) {
+	input := &dynamodb.GetItemInput{
+		ConsistentRead: aws.Bool(req.Options.Consistency == state.Strong),
+		TableName:      aws.String(d.table),
+		Key: map[string]*dynamodb.AttributeValue{
+			"key": {
+				S: aws.String(req.Key),
+			},
+		},
+	}
+
+	result, err := d.client.GetItem(input)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(result.Item) == 0 {
+		return &state.GetResponse{}, nil
+	}
+
+	var output string
+	if err = dynamodbattribute.Unmarshal(result.Item["value"], &output); err != nil {
+		return nil, err
+	}
+	return &state.GetResponse{
+		Data: []byte(output),
+	}, nil
+}
+
+// Set saves a dynamoDB item
+func (d *StateStore) Set(req *state.SetRequest) error {
+	item := map[string]*dynamodb.AttributeValue{
+		"key": {
+			S: aws.String(req.Key),
+		},
+		"value": {
+			S: aws.String(fmt.Sprintf("%v", req.Value)),
```

Comment:
I wonder how this would work for other types too. Double, boolean, byte[] and even JSON (since DynamoDB has native support for JSON). This is tricky in the unit test but would be good to know if all those would work with this implementation.

Reply:
@sayboras [Redis implementation](https://github.com/dapr/components-contrib/blob/37344917183400724f1382a3b314a1ed7d720b5b/state/redis/redis.go#L203-L208) is always serializing req.Value. dapr runtime will pass req.value object to this. it would be good if you can check the type of object and then save it differently. otherwise, let's serialize req.value like redis implementation. 

cc/ @yaron2  @artursouza

Reply:
The only way out for this might be to save it as byte[]. We cannot determine what is the format and structured used in req.Value. It might be JSON or not.